### PR TITLE
from_cmdstanpy - fix bug introduced by refactor (PR 1558)

### PR DESCRIPTION
**Pull request by mitzimorris**
## Description

Refactor of `from_cmdstanpy` failed to account for difference in xarray data shape for scalar and multi-dim vars;
need to squeeze scalar vars to single column.

## Checklist

- [x] Follows [official](https://github.com/arviz-devs/arviz/blob/main/CONTRIBUTING.md#pull-request-checklist) PR format

### CODE REVIEW

**Review by OriolAbril — APPROVED** [review]
Looks great, thanks! 

The documentation build is failing, I think it's due to the latest release of myst-parser, we should pin it to the previous version until it's fixed.

**Comment by mitzimorris** [comment]
why did PR pass before I added shape checks?
did release of parser happen between time of first PR and when I added more checks?

**Comment by OriolAbril** [comment]
The check that failed is the one that makes sure the documentation can be generated, it has no relation to this PR. The checks about the shape did pass (they are evaluated in ExternalTests job, the one that failed is BuildDocs), sorry about that. I have just checked and something must have gone wrong with their release process. They released 0.13.4 right in between these changes above but 0.13.5 is already available and .4 has been yanked from pypi: https://pypi.org/project/myst-parser/#history

Adding the shape checks again now should work and all ci tests should pass

**Comment by mitzimorris** [comment]
tests are failing pylint:  E1133, E1101. - pystan and base_tests

```
/usr/bin/bash --noprofile --norc /home/vsts/work/_temp/4cb9bbfc-22f9-4d96-9fb1-f14175bd8e16.sh
************* Module arviz.data.io_pystan
arviz/data/io_pystan.py:245:23: E1133: Non-iterable value names is used in an iterating context (not-an-iterable)
arviz/data/io_pystan.py:465:23: E1133: Non-iterable value names is used in an iterating context (not-an-iterable)
************* Module arviz.tests.base_tests.test_rcparams
arviz/tests/base_tests/test_rcparams.py:268:8: E1101: Instance of 'dict' has no 'posterior' member (no-member)
arviz/tests/base_tests/test_rcparams.py:268:8: E1101: Instance of 'InferenceData' has no 'posterior' member (no-member)
arviz/tests/base_tests/test_rcparams.py:275:8: E1101: Instance of 'InferenceData' has no 'posterior' member (no-member)
arviz/tests/base_tests/test_rcparams.py:275:8: E1101: Instance of 'dict' has no 'posterior' member (no-member)


```

File: .pylintrc [hunk 1]
```diff
@@ -70,7 +70,9 @@ disable=missing-docstring,
         unsubscriptable-object,
         cyclic-import,
         ungrouped-imports,
-
+	not-an-iterable
+	no-member
```
Comment (OriolAbril) [inline .pylintrc:74]:
```suggestion
	    not-an-iterable
	    no-member

```

**Pull request merged**